src/cdk-experimental/testing/test-app/protractor.e2e.spec.ts

```diff
@@ -0,0 +1,183 @@
+      expect(await items[1].text()).toEqual('TestBed');
+      expect(await items[2].text()).toEqual('Other');
+    });
+
+    it('should be able to locate all sub harnesses', async () => {
+      const alllists = await harness.allLists();
+      const items1 = await alllists[0].getItems();
+      const items2 = await alllists[1].getItems();
+      expect(alllists.length).toEqual(2);
+      expect(items1.length).toEqual(3);
+      expect(await items1[0].text()).toEqual('Protractor');
+      expect(await items1[1].text()).toEqual('TestBed');
+      expect(await items1[2].text()).toEqual('Other');
+      expect(items2.length).toEqual(3);
+      expect(await items2[0].text()).toEqual('Unit Test');
+      expect(await items2[1].text()).toEqual('Integration Test');
+      expect(await items2[2].text()).toEqual('Performance Test');
+    });
+  });
+
+  describe('Test element ', () => {
+    it('should be able to clear', async () => {
+      const input = await harness.input();
+      await input.sendKeys('Yi');
+      expect(await input.getAttribute('value')).toEqual('Yi');
+
+      await input.clear();
+      expect(await input.getAttribute('value')).toEqual('');
+    });
+
+    it('should be able to click', async () => {
+      const counter = await harness.counter();
+      expect(await counter.text()).toEqual('0');
+      await harness.increaseCounter(3);
+      expect(await counter.text()).toEqual('3');
+    });
+
+    it('should be able to send key', async () => {
+      const input = await harness.input();
+      const value = await harness.value();
+      await input.sendKeys('Yi');
+
+      expect(await input.getAttribute('value')).toEqual('Yi');
+      expect(await value.text()).toEqual('Input:Yi');
+    });
+
+    it('focuses the element before sending key', async () => {
+      const input = await harness.input();
+      await input.sendKeys('Yi');
+      expect(await input.getAttribute('id'))
+        .toEqual(await browser.driver.switchTo().activeElement().getAttribute(
+          'id'));
+    });
+
+    it('should be able to hover', async () => {
+      const host = await harness.host();
+      let classAttr = await host.getAttribute('class');
+      expect(classAttr).not.toContain('hovering');
+      await host.hover();
+      classAttr = await host.getAttribute('class');
+      expect(classAttr).toContain('hovering');
+    });
+
+    it('should be able to getAttribute', async () => {
+      const memoStr = `
+        This is an example that shows how to use component harness
+        You should use getAttribute('value') to retrieve the text in textarea
+      `;
+      const memo = await harness.memo();
+      await memo.sendKeys(memoStr);
+      expect(await memo.getAttribute('value')).toEqual(memoStr);
+    });
+
+    it('should be able to getCssValue', async () => {
+      const title = await harness.title();
+      expect(await title.getCssValue('height')).toEqual('50px');
+    });
+  });
+
+  describe('Async operation ', () => {
```

Comment:
Extra whitespace at the end. There's also some a bit below on the other `describe`.